figures/pipeline/enrollment_metrics.py

```diff
@@ -144,17 +152,14 @@ def collect_metrics_for_enrollment(course_enrollment, date_for=None, **_kwargs):
     """
     if not date_for:
         date_for = datetime.utcnow().replace(tzinfo=utc).date()
-    site = get_site_for_course(course_enrollment.course_id)
-    if not site:
-        raise UnlinkedCourseError('No site found for course "{}"'.format(
-            course_enrollment.course_id))
 
     # The following are two different ways to avoide the dreaded error
     #     "Instance of 'list' has no 'order_by' member (no-member)"
     # See: https://github.com/PyCQA/pylint-django/issues/165
-    student_modules = student_modules_for_course_enrollment(ce=course_enrollment)
+    student_modules = course_sm.filter(
+        student_id=course_enrollment.user.id).order_by('-modified')
     if student_modules:
-        most_recent_sm = student_modules.latest('modified')
+        most_recent_sm = student_modules[0]
```

Comment:
How many records are we filtering from?

Reply:
@OmarIthawi The top five counts are 398, 382, 309, 309, 308.

As far as limiting the queries, AFAIK, there's no point in doing a "LIMIT 5" because querysets are lazy evaluated unless you use the "step" parameter of python slicing. Please see here: https://docs.djangoproject.com/en/1.11/topics/db/queries/#limiting-querysets